core/drivers/pinctrl/pinctrl.c

```diff
@@ -96,11 +96,12 @@ TEE_Result pinctrl_get_state_by_idx(const void *fdt, int nodeoffset,
 
 	state->conf_count = conf_count;
 	for (conf_id = 0; conf_id < conf_count; conf_id++) {
-		state->confs[conf_id] =
-			dt_driver_device_from_node_idx_prop(prop_name, fdt,
-							    nodeoffset, conf_id,
-							    DT_DRIVER_PINCTRL,
-							    &res);
+		struct pinconf **pinconf = state->confs + conf_id;
+
+		res = dt_driver_device_from_node_idx_prop(prop_name, fdt,
+							  nodeoffset, conf_id,
+							  DT_DRIVER_PINCTRL,
+							  (void **)pinconf);
```

Comment:
This cast might be a problem, depending on the compiler. It may not be a problem in practice since all pointers use the same size, but a zealous compiler could complain about strict aliasing rules. I guess this is the reason why this somewhat backward way of returning the error value was chosen.

I'd prefer
``` C
                void *pinconf = NULL;

                res = dt_driver_device_from_node_idx_prop(prop_name, fdt,
                                                          nodeoffset, conf_id,
                                                          DT_DRIVER_PINCTRL,
                                                          &pinconf);
                 if (res) {
                        free(state);
                        return res;
                 }
                 state->confs[conf_id] = pinconf;
```
to avoid that ugly cast.